plugins/resources/replication/librepl.cpp

```diff
@@ -378,50 +416,177 @@ extern "C" {
         return result;
     }
 
+    irods::error create_replication_list(
+        irods::resource_plugin_context& _ctx,
+        const std::string& operation) {
+
+        // Get selected hierarchy string
+        std::string selected_hier;
+        std::string root_resc;
+        irods::error ret = get_selected_hierarchy(_ctx, selected_hier, root_resc);
+        if (!ret.ok()) {
+            return PASSMSG(
+                       (boost::format(
+                        "[%s] - Failed to get selected hierarchy.") %
+                        __FUNCTION__).str(), ret);
+        }
+
+        // Get resource name and chop down parser until hierarchy represent this resource
+        irods::hierarchy_parser selected_hier_parser;
+        selected_hier_parser.set_string(selected_hier);
+        std::string name;
+        ret = _ctx.prop_map().get<std::string>(irods::RESOURCE_NAME, name);
+        if (!ret.ok()) {
+            return PASS(ret);
+        }
+        std::string current_resc_hier;
+        ret = selected_hier_parser.str(current_resc_hier, name);
+        if (!ret.ok()) {
+            return PASSMSG(
+                       (boost::format(
+                        "[%s] - Failed to get hierarchy string for resc name [%s].") %
+                        __FUNCTION__ % name).str(), ret);
+        }
+        irods::hierarchy_parser current_hier_parser;
+        current_hier_parser.set_string(current_resc_hier);
+        //rodsLog(LOG_NOTICE, "[%s] - pared down hier to [%s]", __FUNCTION__, current_resc_hier.c_str());
+
+        // Get current hostname for vote
+        char host_name_str[MAX_NAME_LEN];
+        if (gethostname( host_name_str, MAX_NAME_LEN ) < 0) {
+            return ERROR(SYS_GET_HOSTNAME_ERR, "failed in gethostname");
+        }
+        const std::string host_name(host_name_str);
+
+        // Loop over children and create a list of hierarchies to which objects will be replicated
+        float out_vote;
+        child_list_t repl_vector;
+        irods::error result = SUCCESS();
+
+        irods::resource_child_map::iterator it;
+        for ( it = _ctx.child_map().begin(); result.ok() && it != _ctx.child_map().end(); ++it ) {
```

Comment:
Is `result.ok()` necessary in the condition?  It doesn't appear to change in the loop at all.

Reply:
Good point. I believe I based this on the logic found in `replRedirectToChildren` wherein result is updated in the conditional block on 486. result is returned immediately after the loop. We should probably return the `PASSMSG` created there instead of just logging it.